server/src/main/java/com/paypal/selion/grid/servlets/transfer/DefaultCriteria.java

```diff
@@ -0,0 +1,118 @@
+ * artifact folder to uniquely identify a {@link DefaultManagedArtifact}. Parameters artifactName, artifactFolder and
+ * userId match artifact name, folder name and parent folder name of some {@link DefaultManagedArtifact}
+ * respectively.
+ */
+final class DefaultCriteria implements Criteria {
+    
+    private static final SeLionGridLogger LOGGER = SeLionGridLogger.getLogger(DefaultCriteria.class);
+
+    private String artifactName;
+
+    private String uuidFolderName;
+
+    private String subFolderName;
+
+    public DefaultCriteria(Map<String, String> parametersMap) {
+        validateParametersMap(parametersMap);
+        this.artifactName = parametersMap.get(ManagedArtifact.ARTIFACT_FILE_NAME);
+        this.uuidFolderName = parametersMap.get(DefaultRequestParameters.USER_ID);
+        this.subFolderName = parametersMap.get(ManagedArtifact.ARTIFACT_FOLDER_NAME);
+    }
+
+    private void validateParametersMap(Map<String, String> parametersMap) {
+        if (!parametersMap.containsKey(ManagedArtifact.ARTIFACT_FILE_NAME)
+                || !parametersMap.containsKey(DefaultRequestParameters.USER_ID)) {
+            throw new ArtifactDownloadException("Request missing essential parameters: "
+                    + ManagedArtifact.ARTIFACT_FILE_NAME + ", " + DefaultRequestParameters.USER_ID);
+        }
+    }
+
+    public String getArtifactName() {
+        return artifactName;
+    }
+
+    public String getUUIDFolderName() {
+        return uuidFolderName;
+    }
+
+    public String getArtifactFolder() {
+        return subFolderName;
+    }
+
+    public Map<String, String> asMap() {
+        LOGGER.entering();
+        Map<String, String> contentMap = new HashMap<>();
+        contentMap.put(ManagedArtifact.ARTIFACT_FILE_NAME, getArtifactName());
+        contentMap.put(DefaultRequestParameters.USER_ID, getUUIDFolderName());
+        if (!StringUtils.isBlank(getArtifactFolder())) {
+            contentMap.put(ManagedArtifact.ARTIFACT_FOLDER_NAME, getArtifactFolder());
+        }
+        LOGGER.exiting(contentMap);
+        return contentMap;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        if (this == other) {
+            return true;
+        }
+        if (!(other instanceof DefaultCriteria)) {
+            return false;
+        }
+        DefaultCriteria otherCriteria = DefaultCriteria.class.cast(other);
+        if (!getArtifactName().equals(otherCriteria.getArtifactName())) {
+            return false;
+        }
+        if (!getUUIDFolderName().equals(otherCriteria.getUUIDFolderName())) {
+            return false;
+        }
+        boolean equals = getArtifactFolder() == null ? otherCriteria.getArtifactFolder() == null
+                : getArtifactFolder().equals(otherCriteria.getArtifactFolder());
+        if (equals == false) {
+            return false;
+        }
+        return true;
```

Comment:
This was pre-existing and I believe the intent was to check for equality on two instances of `DefaultCriteria` which are pointing to the same meta info. Honestly, since Criteria only come into play for download means, I don't think this is needed.